src/Microsoft.Framework.FileSystemGlobbing/Internal/PathSegments/LiteralPathSegment.cs

```diff
@@ -29,7 +33,7 @@ public override bool Equals(object obj)
             var other = obj as LiteralPathSegment;
 
             return other != null &&
-                string.Equals(other.Value, Value, StringComparison.Ordinal);
+                string.Equals(other.Value, Value, _comparisonType);
```

Comment:
This pattern is pretty strange. I can now create two LiteralPathSegments with different StringComparisons which can cause a.Equals(b) != b.Equals(a).